crates/pallet-bridge-pot-currency-swap/src/genesis_verifier.rs

```diff
@@ -0,0 +1,53 @@
+//! An implementation that verifies bridge pot balance values at genesis.
+
+use frame_support::{
+    sp_runtime::{
+        traits::{CheckedAdd, Convert, Zero},
+        ArithmeticError, DispatchError,
+    },
+    sp_std::marker::PhantomData,
+    traits::{Currency, Get},
+};
+
+use super::{Config, GenesisVerifier, Pallet};
+
+/// The implementation that requires bridge pot balance values be balanced.
+pub struct Balanced<Pallet>(PhantomData<Pallet>);
+
+impl<T: Config<I>, I: 'static> Balanced<Pallet<T, I>> {
+    /// A function to calculate expected [`Config::PotTo`] balance based on the provided list of
+    /// all [`Config::AccountIdFrom`] balances except [`Config::PotFrom`] balance.
+    pub fn calculate_expected_to_bridge_balance(
+        from_balances: &[<T::CurrencyFrom as Currency<T::AccountIdFrom>>::Balance],
+    ) -> Result<<T::CurrencyTo as Currency<T::AccountIdTo>>::Balance, DispatchError> {
```

Comment:
```suggestion
    pub fn calculate_expected_to_bridge_balance<Iter: IntoIter<Item = <T::CurrencyFrom as Currency<T::AccountIdFrom>>::Balance>>(
        from_balances: Iter,
    ) -> Result<<T::CurrencyTo as Currency<T::AccountIdTo>>::Balance, DispatchError> {
```

Reply:
Do we have to use a `DispatchError` here? This is not a call, and we could have a more precise error here...

Reply:
Great suggestion - https://github.com/humanode-network/humanode/pull/700/commits/14909bec64702376f433530bc54726aea4a1b05b